framework/TSA/TimeSeriesAnalyzer.py

```diff
@@ -109,6 +112,30 @@ def getResidual(self, initial, params, pivot, settings):
     residual = initial - sample
     return residual
 
+  def writeXML(self, writeTo, params):
+    """
+      Allows the engine to put whatever it wants into an XML to print to file.
+      @ In, writeTo, xmlUtils.StaticXmlElement, entity to write to
+      @ In, params, dict, parameters from training this ROM
+      @ Out, None
+    """
+    pass # overwrite in subclasses if desired
+
+
+class TimeSeriesGenerator(TimeSeriesAnalyzer):
+  """
+  Act as an identifying class for algorithms that can generate synthetic time histories.
+  """
+
+  def __init__(self, *args, **kwargs):
+    """
+      A constructor that will appropriately intialize a supervised learning object
+      @ In, args, list, an arbitrary list of positional values
+      @ In, kwargs, dict, an arbitrary dictionary of keywords and values
+      @ Out, None
+    """
+    self.name = self.__class__.__name__ # the name the class shall be known by during its RAVEN life
```

Comment:
This appears to be common to both subclasses of TSA; can it remain in the base class?

Reply:
Yes, I believe so.